lib/oscap_tests.pm

```diff
@@ -123,186 +711,541 @@ sub upload_logs_reports {
+        }
+    }
+    else {
+        record_info("Do not modify DS or Ansible files", "Do not modify DS or Ansible files because remove_rules_missing_fixes = $remove_rules_missing_fixes");
+    }
+    backup_ds_file();
 }
 
-sub oscap_remediate {
-    my ($self, $f_ssg_ds, $profile_ID) = @_;
+=ansible return codes
+0 = The command ran successfully, without any task failures or internal errors.
+1 = There was a fatal error or exception during execution.
+2 = Can mean any of:
+
+    Task failures were encountered on some or all hosts during a play (partial failure / partial success).
+    The user aborted the playbook by hitting Ctrl+C, A during a pause task with prompt.
+    Invalid or unexpected arguments, i.e. ansible-playbook --this-arg-doesnt-exist some_playbook.yml.
+    A syntax or YAML parsing error was encountered during a dynamic include, i.e. include_role or include_task.
 
+3 = This used to mean “Hosts unreachable” per TQM, but that seems to have been redefined to 4. I’m not sure if this means anything different now.
+4 = Can mean any of:
+
+    Some hosts were unreachable during the run (login errors, host unavailable, etc). This will NOT end the run early.
+    All of the hosts within a single batch were unreachable- i.e. if you set serial: 3 at the play level, and three hosts in a batch were unreachable. This WILL end the run early.
+    A synax or parsing error was encountered- either in command arguments, within a playbook, or within a static include (import_role or import_task). This is a fatal error. 
+
+5 = Error with the options provided to the command
+6 = Command line args are not UTF-8 encoded
+8 = A condition called RUN_FAILED_BREAK_PLAY occurred within Task Queue Manager.
+99 = Ansible received a keyboard interrupt (SIGINT) while running the playbook- i.e. the user hits Ctrl+c during the playbook run.
+143 = Ansible received a kill signal (SIGKILL) during the playbook run- i.e. an outside process kills the ansible-playbook command.
+250 = Unexpected exception- often due to a bug in a module, jinja templating errors, etc.
+255 = Unknown error, per TQM.
+=cut
+
+sub oscap_remediate {
+    my ($self) = @_;
+    my $out_ansible_playbook;
     select_console 'root-console';
 
     # Verify mitigation mode
-    my $ret
-      = script_run("oscap xccdf eval --profile $profile_ID --remediate --oval-results --report $f_report $f_ssg_ds > $f_stdout 2> $f_stderr", timeout => 600);
-    record_info("Return=$ret", "# oscap xccdf eval --profile $profile_ID --remediate\" returns: $ret");
-    if ($ret != 0 and $ret != 2) {
-        record_info('bsc#1194676', 'remediation should be succeeded', result => 'fail');
+    # For pkg install rules need to refresh repositories
+    # zypper_call('ref -s', timeout => 180);
+    if ($remediated == 0) {
+        push(@test_run_report, "[tests_results]");
     }
+    # If doing ansible playbook remediation
+    if ($ansible_remediation == 1) {
```

Comment:
Do not agree here. oscap_remediate function is complex because of multiple reasons:
1. 2 types of remediations
2. Complex ansible remediation (2 sequential runs) and additional analysis in case of unsuccessful ansible playbook execution.
3. Most of analysis moved out to functions:   ansible_result_analysis, ansible_failed_tasks_search_vv, find_ansible_cce_by_task_name_vv